docs: bump all documentation dependencies

onerandomusername opened the pull request:
## What Changed

Bump all documentation dependencies, restrict them to python 3.11, bump the entire documentation to 3.11 too. 

### Changelogs

- sphinx
    - https://www.sphinx-doc.org/en/master/changes/8.2.html
    - https://www.sphinx-doc.org/en/master/changes/8.1.html
    - https://www.sphinx-doc.org/en/master/changes/8.0.html
    - https://www.sphinx-doc.org/en/master/changes/7.4.html
    - https://www.sphinx-doc.org/en/master/changes/7.3.html
    - https://www.sphinx-doc.org/en/master/changes/7.2.html
    - https://www.sphinx-doc.org/en/master/changes/7.1.html
- sphinx-autobuild
    - https://github.com/sphinx-doc/sphinx-autobuild/compare/2021.03.14...2025.08.25
- sphinx-hoverxref
    - https://github.com/readthedocs/sphinx-hoverxref/compare/1.3.0...1.4.2
- sphinx-notfound-page
  - https://github.com/readthedocs/sphinx-notfound-page/compare/0.8.2...1.1.0
- sphinxcontrib-towncrier
    - https://github.com/sphinx-contrib/sphinxcontrib-towncrier/releases/tag/v0.5.0a0
    - https://github.com/sphinx-contrib/sphinxcontrib-towncrier/releases/tag/v0.4.0a0
- sphinxcontrib-trio
    - no updates
- towncrier
    - https://towncrier.readthedocs.io/en/stable/release-notes.html `23.6.0` -> `25.8.0`

### Deprecations

sphinx-hoverxref uses some interfaces removed in sphinx 9.0. However, sphinx-hoverxref is it itself deprecated, so we should drop it anyhow.

## Depends on
#1341, which limits docs to only being typechecked on the version it runs on.

shiftinv commented:
Are you planning on dropping hoverxref in this or in a separate PR?

onerandomusername commented:
> Are you planning on dropping hoverxref in this or in a separate PR?

Planning to drop hoverxref on another PR.

This one is also blocked by needing to figure out pyright 3.8-3.10 overrides

onerandomusername commented on pyproject.toml:
```diff
@@ -85,6 +75,16 @@ codemod = [
+    "sphinxcontrib-towncrier==0.5.0a0; python_version >= '3.11'",
+    "sphinxcontrib-trio~=1.1.2; python_version >= '3.11'",
+    "sphinxext-opengraph==0.9.1",
+    { include-group = "changelog" },
```
apparently pyright's parser wasn't fixed for this until 1.1.387, so bumping pyright is necessary to dedupe dependencies.

onerandomusername committed:
deps: bump doc dependencies

onerandomusername committed:
fix type error

onerandomusername committed:
configure ruff for py311 docs

onerandomusername committed:
Merge branch 'master' into docs/bump-sphinx-8

onerandomusername committed:
deps: bump sphinxext-opengraph

onerandomusername committed:
fix: don't install all dependencies with the dev action

onerandomusername committed:
bump towncrier

onerandomusername committed:
apply suggestions

shiftinv commented:
A couple more things I noticed:
- Sphinx added the classic purple coloring to visited links, the contrast with the documentation's background color is *awful* (contrast ratio of 1.05): 
  <img width="671" height="184" alt="image" src="https://github.com/user-attachments/assets/5796b899-05d0-49f1-b802-b6393f1983b8" />
  I think the best solution would be to override that selector and removing the highlight, visited link tracking for a documentation website seems fairly pointless to me anyway.
- Maybe it's just the PR preview build, but search is pretty broken outside of the indexed results; it requests pages like `https://disnake--1325.org.readthedocs.build/en/1325/undefinedext/tasks/index.html`, which obviously don't resolve. Haven't really investigated why, yet.

(also, thanks for including links to all the changelogs <3)

shiftinv committed:
fix: replace obsolete `URL_ROOT` with html data field for search

shiftinv committed:
fix: override/reset :visited highlight

shiftinv committed:
fix: remove duplicate `documentation_options.js` script

see https://github.com/sphinx-doc/sphinx/commit/ee5560b7739932d0a7735be0cc81f28d76a61e92

shiftinv committed:
chore: rename `.js_t` to `.js.jinja`

https://www.sphinx-doc.org/en/master/development/html_themes/index.html#static-templates

shiftinv committed:
fix: suppress two verbose/unimportant sphinx logs

shiftinv committed:
fix: ignore `__pycache__` when autobuilding, to avoid building twice

shiftinv committed:
fix: base custom search scorer on base title score

this is by no means perfect; long story short, sphinx changed
the search scorer algorithm to actually take the configured `title`
score (15) into account for matching titles, instead of a fixed 100.
This, in turn, meant that any and all page matches would now be ranked at the bottom,
since the custom search scorer inflates the score of all `disnake.*` objects to
a score right below 100.

To resolve this, the scorer now assigns these API results a score of
14-15, i.e. right below pages with matching titles again.
API results with a longer or later submatch continue to receive a higher score penalty,
and are therefore closer to 14 rather than 15.

shiftinv committed:
chore: comment debug console.log again

shiftinv committed:
Merge remote-tracking branch 'upstream/master' into docs/bump-sphinx-8

shiftinv committed:
chore: reorder pyproject keys a little

shiftinv committed:
revert: undo scorer changes for now

shiftinv committed:
revert: reset custom scorer to sphinx defaults

shiftinv committed:
docs: implement new custom search scorer

For the most part, this scores things identically to Sphinx's builtin
scorer, with two notable differences:

- object name substring matches (outside the last dotted part) are ranked
  higher than fulltext matches, since the latter are fairly useless in most cases
- for full name matches (i.e. either exact match or at least exact match of last dotted part),
  results are reordered to show shorter (by dotted parts) results first, in an attempt to show
  e.g. `disnake.Client` before `disnake.ApplicationCommandInteraction.client`, since those would
  otherwise just be ordered alphabetically, while someone searching for "Client" would most likely
  be more interested in the former than the latter

shiftinv committed:
chore(typing): fix list/sequence sillyness

shiftinv committed:
Merge branch 'master' into docs/bump-sphinx-8
